src/server/controlchan/commands/pasv.rs

```diff
@@ -119,10 +125,40 @@ where
             }
         });
 
-        //Ok(Reply::None)
         Ok(Reply::new_with_string(
             ReplyCode::EnteringPassiveMode,
             format!("Entering Passive Mode ({},{},{},{},{},{})", octets[0], octets[1], octets[2], octets[3], p1, p2),
         ))
     }
+
+    // For proxy mode we prepare the session and let the proxy loop know (via channel) that it
+    // should choose a data port and check for connections on it.
+    async fn handle_proxy_mode<S, U>(&self, args: CommandContext<S, U>, mut tx: ProxyLoopSender<S, U>) -> Result<Reply, ControlChanError>
+    where
+        U: UserDetail + 'static,
+        S: 'static + storage::StorageBackend<U> + Sync + Send,
+        S::File: tokio::io::AsyncRead + Send,
+        S::Metadata: storage::Metadata,
+    {
+        tx.send(ProxyLoopMsg::AssignDataPortCommand(args.session.clone())).await.unwrap();
```

Comment:
For proxy mode, this

Reply:
For proxy mode, this is already done in the proxy loop. It does seem better to do it here instead. But then you need to remove the data loop comms part in https://github.com/bolcom/libunftp/blob/hannes/proxy-prot/src/server/ftpserver.rs#L424

Reply:
tbh i don't know why i put this code in the proxy loop in the first place...